File: app/src/main/java/com/asfoundation/wallet/repository/WalletRepositoryType.java
package com.asfoundation.wallet.repository;

import com.asfoundation.wallet.entity.Wallet;
import io.reactivex.Completable;
import io.reactivex.Single;
import java.math.BigDecimal;

public interface WalletRepositoryType {
  Single<Wallet[]> fetchWallets();

  Single<Wallet> findWallet(String address);

  Single<Wallet> createWallet(String password);

  Single<Wallet> restoreKeystoreToWallet(String store, String password, String newPassword);

  Single<Wallet> restorePrivateKeyToWallet(String privateKey, String newPassword);

  Single<String> exportWallet(String address, String password, String newPassword);

  Completable deleteWallet(String address, String password);

  Completable setDefaultWallet(String address);

  Single<Wallet> getDefaultWallet();

  Single<BigDecimal> getEthBalanceInWei(String address);

  Single<BigDecimal> getAppcBalanceInWei(String address);
}

File: app/src/main/java/com/asfoundation/wallet/repository/WalletRepositoryType.kt
package com.asfoundation.wallet.repository

import com.asfoundation.wallet.entity.Wallet
import com.asfoundation.wallet.recover.RecoverWalletResult
import io.reactivex.Completable
import io.reactivex.Observable
import io.reactivex.Single

interface WalletRepositoryType {

  fun fetchWallets(): Single<Array<Wallet>>

  fun findWallet(address: String): Single<Wallet>

  fun createWallet(password: String): Single<Wallet>

  fun restoreKeystoreToWallet(store: String, password: String,
                              newPassword: String): Single<RecoverWalletResult>

  fun restorePrivateKeyToWallet(privateKey: String?, newPassword: String): Single<RecoverWalletResult>

  fun exportWallet(address: String, password: String, newPassword: String?): Single<String>

  fun deleteWallet(address: String, password: String): Completable

  fun setDefaultWallet(address: String): Completable

  fun getDefaultWallet(): Single<Wallet>

  fun observeDefaultWallet(): Observable<Wallet>
}